markus_test_scripts/examples/test_markus_integration.R

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env Rscript
+
+# Load helper functions
+source("r_helpers.R")
+
+# Test basic LLM function with a simple example
+test_basic_functionality <- function() {
+  test_code <- 'x <- 1 + 1\nprint(x)'
+  writeLines(test_code, "test_submission.R")
+  
+  tryCatch({
+    # Test code analysis
+    result <- run_llm_r(
+      prompt = "code_prompt",
+      scope = "code",
+      model = "claude",
+      submission_file = "test_submission.R"
+    )
+    
+    if (is.character(result) && nchar(result) > 0) {
+      cat("Code analysis test passed\n")
+      cat("Response length:", nchar(result), "characters\n")
+    } else {
+      cat("Code analysis test failed - empty response\n")
+    }
+    
+    unlink("test_submission.R")
+    
+  }, error = function(e) {
+    cat("Code analysis test failed:", e$message, "\n")
+    unlink("test_submission.R")
+  })
+}
+
+# Test fixture integration
+test_fixture_integration <- function() {
+  
+  # Test with existing fixtures
+  fixture_path <- "../tests/fixtures/code_example/fail_submission/fail_submission.R"
+  
+  if (file.exists(fixture_path)) {
+    cat("Found test fixture:", fixture_path, "\n")
+    
+    tryCatch({
+      feedback <- run_llm_r(
+        prompt = "code_prompt",
+        scope = "code",
+        model = "claude",
+        submission_file = fixture_path
+      )
+      
+      if (is.character(feedback) && nchar(feedback) > 0) {
+        cat("Fixture integration test passed\n")
+      } else {
+        cat("Fixture integration test failed - empty response\n")
+      }
+      
+    }, error = function(e) {
+      cat("Fixture integration test failed:", e$message, "\n")
+    })
+  } else {
+    cat("Test fixture not found, skipping fixture integration test\n")
+  }
+}
+
+# Test QMD image generation fixture
+test_qmd_fixture <- function() {  
+  qmd_fixture_path <- "../tests/fixtures/sta130_example/submission.qmd"
+  
+  if (file.exists(qmd_fixture_path)) {
+    cat("Found QMD fixture:", qmd_fixture_path, "\n")
+    
+    tryCatch({
+      result <- run_llm_r(
+        prompt = "image_prompt",
+        scope = "image",
+        model = "claude", 
+        submission_file = qmd_fixture_path
+      )
+      
+      if (is.character(result) && nchar(result) > 0) {
+        cat("QMD image generation test passed\n")
+      } else {
+        cat("QMD image generation test failed - empty response\n")
+      }
+      
+    }, error = function(e) {
+      cat("QMD image generation test failed:", e$message, "\n")
+    })
+  } else {
+    cat("QMD fixture not found, skipping QMD test\n")
+  }
+}
+
+# Test Markus output formatting
```

Comment:
This test and the one below don't use the AI feedback library, so they are not helpful for the purpose of this file (which, to be clear, is to illustrate how the AI feedback library can be used within MarkUs).